src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -64,6 +65,9 @@ public Pinecone build() {
     }
 
     public IndexModel createIndex(CreateIndexRequest createIndexRequest) throws PineconeException {
+        if (createIndexRequest == null) {
+            throw new PineconeException("CreateIndexRequest object cannot be null");
```

Comment:
We can swap to `PineconeValidationException` for these errors I think, as it's a bit more specific. It's also what we use when throwing out of data plane calls that are missing required fields in `IndexInterface`: https://github.com/pinecone-io/pinecone-java-client/blob/640424f61c2e75067bf2ddcb478481144d4e8da8/src/main/java/io/pinecone/commons/IndexInterface.java#L59

Reply:
It's weird... I have changed it now to `PineconeValidationException` and changed the expected `throw` in the tests, but the tests don't care if I have `PineconeValidationException` as the expected `throw` or `PineconeException`... (but it does care if I mess with the expected message, which is good)

Reply:
I think it's because it extends it: https://github.com/pinecone-io/pinecone-java-client/blob/640424f61c2e75067bf2ddcb478481144d4e8da8/src/main/java/io/pinecone/exceptions/PineconeValidationException.java#L3

Reply:
Yaya deleted my comment, thank yewwww!